### PR TITLE
Bug Fix - Image and Signature Extraction 

### PR DESCRIPTION
**Pull request by Arkajit-Datta**
The following changes must fix the issue regarding saving of Image and extraction of signature from the newer versions of AadharQRs

### CODE REVIEW

**Comment by Tanmoy741127** [comment]
@Arkajit-Datta can you send me the latest docs of secure QR

**Comment by Arkajit-Datta** [comment]
sure @Tanmoy741127

**Comment by srikanthlogic** [comment]
> sure @Tanmoy741127

Is this available publicly?

**Comment by Arkajit-Datta** [comment]
Sorry @Tanmoy741127, I was tied up the lately, will be looking into your comments if changes are required, Thanks!

**Comment by Tanmoy741127** [comment]
Thanks. I will look at that.

**Comment by Tanmoy741127** [comment]
> @srikanthlogic , oh yes it is, let me share the link here, https://uidai.gov.in/images/resource/User_manulal_QR_Code_15032019.pdf

Are you sure this is the newest one ? Because while building the library I have followed this one published on 2019. But you have mentioned that version 2 , 2022 version in code comments

File: build/lib/pyaadhaar/decode.py [hunk 1]
```diff
@@ -11,46 +12,51 @@
...
+    # This class now supports 2022 version of Aadhaar QR codes [version-2]
     # For more information check here : https://103.57.226.101/images/resource/User_manulal_QR_Code_15032019.pdf
 
     def __init__(self, base10encodedstring):
         self.base10encodedstring = base10encodedstring
-        self.details = ["referenceid", "name", "dob", "gender", "careof", "district", "landmark",
-                        "house", "location", "pincode", "postoffice", "state", "street", "subdistrict", "vtc"]
-        self.delimeter = []
+        self.details = ["version","email_mobile_status","referenceid", "name", "dob", "gender", "careof", "district", "landmark",
+                        "house", "location", "pincode", "postoffice", "state", "street", "subdistrict", "vtc", "last_4_digits_mobile_no"]
```
Comment (Tanmoy741127) [inline build/lib/pyaadhaar/decode.py:21]: According to the docs, in the xml there should not be any field for last 4 digits of mobile no. Only mobile no hash will be there to verify with mobile no.
There is a reference id which is 4 digit of your aadhar card no

File: build/lib/pyaadhaar/decode.py [hunk 2]
```diff
@@ -69,49 +75,61 @@ def signedData(self):
...
+                return False
+            else:
+                return True
+        elif int(self.data['email_mobile_status']) == 0:
+    
     def image(self):
         # Will return the image stream to be used in another function
         if int(self.data['email_mobile_status']) == 3:
-            return Image.open(BytesIO(self.decompressed_array[self.delimeter[15]+1:len(self.decompressed_array)-256-32-32]))
+            print("length --> ", len(self.decompressed_array[self.delimeter[len(self.details)]+1:len(self.decompressed_array)]))
```
Comment (Tanmoy741127) [inline build/lib/pyaadhaar/decode.py:120]: remove the print statement, or add a debug value in class to print while debug is set to true

File: build/lib/pyaadhaar/tests.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+from pdf2image import convert_from_path
+from pyzbar.pyzbar import decode
+from decode import AadhaarSecureQr
+
+filename = '/home/arkajit/Documents/Projects/pyaadhaar_updated/test_files/Test_Aadhar_Arkajit.pdf'
```
Comment (Tanmoy741127) [inline build/lib/pyaadhaar/tests.py:5]: Here it is the file path of file in your system
Either write a proper testcase which user can run or drop this file

File: pyaadhaar.egg-info/SOURCES.txt [hunk 4]
```diff
@@ -1,6 +1,7 @@
 setup.py
 pyaadhaar/__init__.py
 pyaadhaar/decode.py
+pyaadhaar/tests.py
```
Comment (Tanmoy741127) [inline pyaadhaar.egg-info/SOURCES.txt:4]: fix the tests.py with proper testcase or remove

**Review by Tanmoy741127 — CHANGES_REQUESTED** [review]
Delete the __pycache__ folders and add it to .gitignore also

File: build/lib/pyaadhaar/decode.py (hunk 1 above)
Comment (Arkajit-Datta) [reply build/lib/pyaadhaar/decode.py:21]: But, while testing it with the new aadhaar cards I found the last 4 digit number is displayed there. These aadhaar cards are the newest ones, though there hasn't been any update in the documentation

File: build/lib/pyaadhaar/decode.py (hunk 2 above)
Comment (Arkajit-Datta) [reply build/lib/pyaadhaar/decode.py:120]: Sure! print statement will be removed

File: build/lib/pyaadhaar/tests.py (hunk 3 above)
Comment (Arkajit-Datta) [reply build/lib/pyaadhaar/tests.py:5]: Sure will put some test cases, but I was worried about the information that'll go into the public domain.

File: pyaadhaar.egg-info/SOURCES.txt (hunk 4 above)
Comment (Arkajit-Datta) [reply pyaadhaar.egg-info/SOURCES.txt:4]: Sure! will add this

File: build/lib/pyaadhaar/decode.py (hunk 1 above)
Comment (Tanmoy741127) [reply build/lib/pyaadhaar/decode.py:21]: please check again, this 4digit nuber should be last 4 digiti of aadhar card. Check and confirm please

File: build/lib/pyaadhaar/tests.py (hunk 3 above)
Comment (Tanmoy741127) [reply build/lib/pyaadhaar/tests.py:5]: You can build a QR from the datastring provided in the Aaadhar Secure QR Docs as example

**Comment by tanmoysrt** [comment]
@Arkajit-Datta great work 🚀
Some changes required, I will do that in followup PR

**Comment by Arkajit-Datta** [comment]
Thanks @tanmoysrt

**Pull request merged**